src/client/mod.rs

```diff
@@ -175,19 +175,29 @@ impl APIClient {
 
                                 // Split on newlines
                                 for line in buff.split(|x| x == &b'\n') {
-                                    match serde_json::from_slice(&line) {
-                                        Ok(val) => items.push(Ok(val)),
-                                        Err(e) if e.is_eof() => {
-                                            // In this case we've received only a partial object
-                                            // so store it for use in the next iteration
-                                            new_buff.extend_from_slice(&line);
+                                    new_buff.extend_from_slice(&line);
+
+                                    match serde_json::from_slice(&new_buff) {
+                                        Ok(val) => {
+                                            // on success clear our buffer
+                                            new_buff.clear();
+                                            items.push(Ok(val));
```

Comment:
This seems theoretically solid. I wish there was a way of testing this. My sleep is being really weird.